### DISCUSSION TITLE
What is the way i could manage my cnpg clusters?

### DISCUSSION THREAD

**Discussion by wwnbb** [cloudnative-pg/cloudnative-pg#2724] [Q&A] [answered]
I'm building micro service architecture based on cnpg, the thing that i struggle with is that in documentation and inside examples i cannot see any references for multiple cnpg deployed among k8s cluster. So the question is: should i deploy separate cluster for each of my app microservice, if i do that is there any rough edges, like how should i monitor each of them?

**Comment by gbartolini** [comment, marked as answer]
Provided I understood correctly your statement, It is really up to you. Our advice is to:

- implement the microservice database, hence one application, one database owned by the application development team
- run a single database (the application database) inside a `Cluster`

If you are using Kubernetes, monitoring should happen on the Kubernetes/infrastructure side, through Prometheus metrics, alerts and a dashboard (like Grafana). The idea is that each team is responsible for their own applications end to end.

I hope this helps.
